fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -2,12 +2,15 @@
 # Licensed under the MIT License.
 
 import logging
+import math
 import numpy as np
 import pandas as pd
 import scipy.optimize as opt
 from sklearn import clone
 from sklearn.dummy import DummyClassifier
 from time import time
+import inspect
+import resampling
```

Comment:
Since `resampling` is part of Fairlearn..... we usually do those imports with a prefixed `.` to emphasise this (like in the `._constants` below). That bit of coding style could be up for debate, though.